Fixes - #4 Add test cases for client side sdk

siddharthashw opened the pull request:
Fixes - #4 

**1. IStringConstant.cs** - added string constant for test cases
**2. StringConstant.cs** - added string constant for test cases
**3. BaseProvider.cs** - added to resolve dependency for test case
**4. project.json(Test)** - Added Package
"testRunner": "xunit",
"Microsoft.Extensions.DependencyInjection": "1.2.0-preview1-22929",
"Moq": "4.6.38-alpha",
"Promact.OAuth.Client": "1.0.0-*",
"xunit": "2.2.0-beta2-build3300",
"dotnet-test-xunit": "2.2.0-preview2-build1029"
Change the framework of test project from netstandard1.6 to netcoreapp1.1 and added its dependedncy
**5. ProjectModuleTest.cs** - Added for test cases of project module - added 11 test case for project all scenario
**6. UserModuleTest.cs** - Added for test cases of user module. - added 39 test cases for project all scenario
**7. UserModule** - Inherit IUserModule
**8. ProjectModule** - used using for namespace than extension

siddharthashw committed:
IStringConstant.cs - added string constant for test cases
StringConstant.cs - added string constant for test cases
BaseProvider.cs - added to resolve dependency for test case
project.json(Test) - Added Package
"testRunner": "xunit",
"Microsoft.Extensions.DependencyInjection": "1.2.0-preview1-22929",
"Moq": "4.6.38-alpha",
"Promact.OAuth.Client": "1.0.0-*",
"xunit": "2.2.0-beta2-build3300",
"dotnet-test-xunit": "2.2.0-preview2-build1029"
Change the framework of test project from netstandard1.6 to netcoreapp1.1 and added its dependedncy
ProjectModuleTest.cs - Added for test cases of project module - added 11 test case for project all scenario
UserModuleTest.cs - Added for test cases of user module.

siddharthashw committed:
IStringConstant.cs -  added variables for test case
StringConstant.cs -  added variables for test case
ProjectModuleTest.cs - update region from private method to mocking
UserModuleTest.cs - added 39 test cases for user module
Fixes - #4

rushi216 commented on Promact.OAuth.Client/src/Promact.OAuth.Client/Util/StringConstant/StringConstant.cs:
```diff
@@ -208,5 +208,249 @@ public string AccessTokenNotMatchedException
+        {
+            get
+            {
+                return "[{\"firstName\":\"Siddhartha\",\"lastName\":null,\"isActive\":false,\"numberOfCasualLeave\":0.0,\"numberOfSickLeave\":0.0,\"joiningDate\":\"0001-01-01T00:00:00\",\"slackUserId\":\"U0HJ49KJ4\",\"slackUserName\":null,\"projects\":null,\"createdBy\":null,\"createdDateTime\":\"0001-01-01T00:00:00\",\"updatedBy\":null,\"updatedDateTime\":\"0001-01-01T00:00:00\",\"id\":\"c2e7cb6 ...
```
Why would you put json string for random values? Couldn't you create application class for same and serialize it while using it. It would be better way when you want to change any property in future

siddharthashw replied:
I have used application class for serialization later on method. This is used to return value from mock of http client wrapper.

siddharthashw replied:
Project.cs - updated property type of CreatedDate and UpdatedDate from string to DateTime
IStringConstant.cs - removed json random value string
StringConstant.cs - removed json random value string
ProjectModuleTest.cs - Added Serialization of object and return as string for mocking
UserModuleTest.cs - Added Serialization of object and return as string for mocking

rushi216 reviewed (changes requested):
Please look into the serialized json string

siddharthashw committed:
Project.cs - updated property type of CreatedDate and UpdatedDate from string to DateTime
IStringConstant.cs - removed json random value string
StringConstant.cs - removed json random value string
ProjectModuleTest.cs - Added Serialization of object and return as return for mocking
UserModuleTest.cs - Added Serialization of object and return as return for mocking
